### ISSUE TITLE
Wrong "export PATH" statement for Powershell Core

### ISSUE THREAD

**Issue by Kirides** [golang/vscode-go#378]
### What version of Go, VS Code & VS Code Go extension are you using?
- Run `go version` to get version of Go
	- go version go1.14.2 windows/amd64
- Run `code -v` or `code-insiders -v` to get version of VS Code or VS Code Insiders
	- 1.47.2
17299e413d5590b14ab0340ea477cdd86ff13daf
x64
- Check your installed extensions to get the version of the VS Code Go extension 
	- 2020.7.820 (just ran the `Extension: Check for Extension Updates` command)
- Run `go env` to get the go development environment details
	- set GO111MODULE=
set GOARCH=amd64
set GOBIN=
set GOCACHE=C:\Users\USER\AppData\Local\go-build
set GOENV=C:\Users\USER\AppData\Roaming\go\env  
set GOEXE=.exe
set GOFLAGS=
set GOHOSTARCH=amd64
set GOHOSTOS=windows
set GOINSECURE=
set GONOPROXY=
set GONOSUMDB=
set GOOS=windows
set GOPATH=A:\Dev\Go
set GOPRIVATE=
set GOPROXY=https://proxy.golang.org,direct
set GOROOT=c:\go
set GOSUMDB=sum.golang.org
set GOTMPDIR=
set GOTOOLDIR=c:\go\pkg\tool\windows_amd64
set GCCGO=gccgo
set AR=ar
set CC=gcc
set CXX=g++
set CGO_ENABLED=1
set GOMOD=C:\Users\USER\Desktop\go\langsrv\go.mod
set CGO_CFLAGS=-g -O2
set CGO_CPPFLAGS=
set CGO_CXXFLAGS=-g -O2
set CGO_FFLAGS=-g -O2
set CGO_LDFLAGS=-g -O2
set PKG_CONFIG=pkg-config
set GOGCCFLAGS=-m64 -mthreads -fmessage-length=0 -fdebug-prefix-map=C:\Users\USER\AppData\Local\Temp\go-build460855558=/tmp/go-build -gno-record-gcc-switches

### Share the Go related settings you have added/edited

```js
    "terminal.integrated.shell.windows": "pwsh.exe",
    "terminal.integrated.env.windows": {
        "PATH": "${env:PATH};C:\\msys64\\mingw64\\bin;C:\\msys64\\usr\\bin;c:\\go\\bin"
    },
    "terminal.integrated.shellArgs.windows": [
        "-NoLogo",
    ],
    "go.useLanguageServer": true,
    "[go]": {
        "editor.snippetSuggestions": "top",
        "editor.formatOnSave": true,
        "editor.codeActionsOnSave": {
            "source.organizeImports": true,
        }
    },
    "gopls": {
        "usePlaceholders": true, // add parameter placeholders when completing a function
        // Experimental settings
        "completeUnimported": true, // autocomplete unimported packages
        "deepCompletion": true, // enable deep completion
    },
```

### Describe the bug

Using Powershell Core leads to this:
![grafik](https://user-images.githubusercontent.com/13602143/87881423-74b0b980-c9f9-11ea-8f33-0520f49393ef.png)

The Terminal is started using `pwsh.exe` as seen in configuration above. The name displayed in the integrated terminal switches to `dotnet` after booting up the `pwsh.exe`

Issue is simmilar to #314 

### Steps to reproduce the behavior:
1. Set `pwsh.exe` (powershell core) as terminal
2. Open the integrated terminal
3. observe the mentioned error

**Comment by hyangah** [comment]
@Kirides thanks for the report. It's a duplicate of #314. Due to an issue in our nightly release workflow https://github.com/golang/vscode-go/issues/363 we didn't publish a new Nightly version that includes the fix yet.

**Comment by Kirides** [comment]
While that's true, i don't see the same special handling for `pwsh` in the committed code that closed #314

**Comment by hyangah** [comment]
@Kirides thanks for the headsup. 
@mcjcloud - looking at https://github.com/microsoft/vscode/issues/74233#issuecomment-497527337, I think we need adjustment in shell detection. https://github.com/microsoft/vscode-python/blob/649156a09ccdc51c0d20f7cd44540f1918f9347b/src/client/common/terminal/shellDetector.ts#L24-L35 is a good starting point. And, for unrecognized shell types, let's just give up and do nothing.

**Comment by mcjcloud** [comment]
I've created a change which will do two of the above items: add `pwsh` as a recognized shell name and prevent any `export` command from being run as a fallback when no shell is recognized.

I'm leaving this issue open to track progress on improving the integrated terminal detection further, starting with what @hyangah mentioned (https://github.com/microsoft/vscode-python/blob/649156a09ccdc51c0d20f7cd44540f1918f9347b/src/client/common/terminal/shellDetector.ts#L24-L35).

**Comment by hyangah** [comment]
https://github.com/golang/vscode-go/issues/146#issuecomment-662885510 @segevfiner suggested `ctx.environmentVariableCollection` which looks promising. I am not yet sure if this also applies to the terminal already open before the extension determines the GOPATH, but in worst case, the extension can suggest close/open if the extension can't do it itself. 

cc @mcjcloud

**Comment by segevfiner** [comment]
@hyangah It is supposed to come with handling for those race conditions.

**Comment by hyangah** [comment]
https://github.com/microsoft/vscode/issues/46696 is the overview of the API history,
https://github.com/microsoft/vscode-extension-samples/blob/2e5b265e00417dde0fea453eb8abdd34a8ec7d2c/terminal-sample/src/extension.ts is the sample.

I am still struggling with the interaction with the terminal login shell. 
After prepending the desirable go path to the existing `PATH`, vscode suggested to relaunch the terminal.

Then, when relaunching the terminal, it seems like the integrated terminal runs `.bash_profile` again
where I have various PATH prepending again. 
As a result, the go path I prepended ends up in the middle of the PATH. :-(

**Comment by hyangah** [comment]
Maybe this https://github.com/microsoft/vscode/issues/99878

**Comment by segevfiner** [comment]
Also note that the Python extension does a similar thing for selecting Python/virtualenv so its something to look at. I actually won't be suprised if that API was created for it. But it might indeed still have issues, I'm not sure if the Python extension is using it yet either.

**Comment by hyangah** [comment]
Fixes for this is included in v0.16.0-rc.1
https://github.com/golang/vscode-go/releases/tag/v0.16.0-rc.1 

To install the rc version, download the vsix file from the release page, 
and from VSCode Command Palette -> 'Extensions: Install from VSIX...', and choose the downloaded vsix file.
